var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -491,6 +498,27 @@ def patch(self):
             "torch_global_deps PROPERTIES LINKER_LANGUAGE CXX",
             "caffe2/CMakeLists.txt",
         )
+        if self.spec.satisfies("@2.1: + rocm"):
+            filter_file(
+                r"${ROCM_INCLUDE_DIRS}/rocm_version.h",
+                "{0}/include/rocm-core/rocm_version.h".format(self.spec["rocm-core"].prefix),
+                "cmake/public/LoadHIP.cmake",
+                string=True,
+            )
+            filter_file(
+                r"-DINCLUDE_DIRECTORIES=${ROCM_INCLUDE_DIRS}",
+                "-DINCLUDE_DIRECTORIES={0}/include/rocm-core".format(
+                    self.spec["rocm-core"].prefix
+                ),
+                "cmake/public/LoadHIP.cmake",
+                string=True,
+            )
+            filter_file(
+                r"__HIP_PLATFORM_HCC__",
+                "__HIP_PLATFORM_AMD__",
+                "caffe2/CMakeLists.txt",
+                string=True,
+            )
```

Comment:
Same question here, has this been fixed upstream so it will no longer be needed someday?

Reply:
PR  [2001](https://github.com/ROCm/pytorch/pull/2001) in review to add these include paths.

Reply:
By upstream I mean https://github.com/pytorch/pytorch

Reply:
Our py-torch team needs the changes in release/ or internal_testing branches before getting this in the https://github.com/pytorch/pytorch repo.